### ISSUE TITLE
Failing with cachier 3.1.1

### ISSUE THREAD

**Issue by veenstrajelmer** [mercator-ocean/copernicus-marine-toolbox#183]
When running copernicusmarine with cachier 3.1.1 the following error is given:
![image](https://github.com/user-attachments/assets/ddc76167-db65-47d3-8141-eda852271ff9)

This happens with cachier 3.1.1, but it works just fine with cachier 3.0.1 (their previous release)

**TODO**: a reproducible example will follow

**Comment by renaudjester** [comment]
@veenstrajelmer Thanks for the reporting! 

I guess I can limit the version of cachier library if there are no workaround!

Also in the v2 we are preparing cachier is not used by the toolbox 😄

**Comment by veenstrajelmer** [comment]
Great, that would be amazing!

**Comment by Leinadium** [comment]
I found the same problem when using `copernicusmarine` and reported on the `cachier` repository, but not here!
I'm working on submitting a PR to fix cachier's bug, but using ` cachier==3.0.1` seems to be working.

**Comment by Leinadium** [comment]
I [suspect](https://github.com/python-cachier/cachier/issues/246#issuecomment-2432232733) the cache files from `cachier` are incompatible between versions. Both are pickle files, but one uses dicts, and the newer one uses dataclasses.

If this is true, purging your local cache at `~/.copernicusmarine/cache` max fix this problem.

**Comment by renaudjester** [comment]
Indeed it seems that it's a compatibility problem between the versions.

If I install the toolbox (from a virgin environment) then `describe` works and `cachier==3.1.1`.

If I then I change to `cachier==3.0.1` then `describe` breaks with the error: 
```
Attribute error: Can't get attribute 'CacheEntry' on <module 'cachier.config' from '/.../micromamba/envs/copernicusmarine-test/lib/python3.12/site-packages/cachier/config.py'>
```

If I use the `--overwrite-metadata-cache` then it works again, it fixes the problem. 

If I then change to `cachier==3.1.1` then `describe` breaks with the error (the one you described): 
```
Attribute error: 'dict' object has no attribute '_completed'
```

Again it can be fixed with `--overwrite-metadata-cache` . 

I don't know if it requires a fix 🤔 Knowing that this happens anyway with cachier in many situation (that the cache is not compatible anymore eg when changing python version)

**Comment by veenstrajelmer** [comment]
Interesting, thanks for investigating this. I guess you are right, especially since you mention cachier will not be used in v2 anymore.  However, this error is quite confusing, maybe it can be caught and then tell the user to overwrite the cache?

**Comment by renaudjester** [comment]
A more comprehensive message and wider retry will be implemented in the v1.3.4 version :D Thanks for the reporting @veenstrajelmer  and @Leinadium

**Comment by veenstrajelmer** [comment]
Thanks a lot for fixing this! I noticed also there was a fix considered in cachier (discussion in https://github.com/python-cachier/cachier/issues/246), but great that copernicusmarine also covers this!

**Comment by shaypal5** [comment]
Done. A fix was released in `v3.1.2`. Please let us know if this is not the case, and happy caching to the lot of you!
